### PR TITLE
Make ChangeArrayPushToArrayAssignRector use stmts

### PR DESCRIPTION
**Pull request by TomasVotruba**

### CODE REVIEW

File: rules/CodeQuality/Rector/FuncCall/ChangeArrayPushToArrayAssignRector.php [hunk 1]
```diff
@@ -59,61 +46,54 @@ public function run()
...
-        $parent = $node->getAttribute(AttributeKey::PARENT_NODE);
-        if (! $parent instanceof Expression) {
+        if ($this->hasArraySpread($funcCall)) {
             return null;
         }
 
-        if (! isset($node->args[0])) {
-            return null;
-        }
+        $args = $funcCall->getArgs();
```
Comment (samsonasik) [inline rules/CodeQuality/Rector/FuncCall/ChangeArrayPushToArrayAssignRector.php:71]: `$args === []` need to be checked early, return null

Comment (samsonasik) [reply rules/CodeQuality/Rector/FuncCall/ChangeArrayPushToArrayAssignRector.php:71]: @TomasVotruba empty args is allowed while show warning in php 7.4, ref https://3v4l.org/TEJCb#v740 so empty args needs to be checked.

Comment (TomasVotruba) [reply rules/CodeQuality/Rector/FuncCall/ChangeArrayPushToArrayAssignRector.php:71]: I see. Could you fix it in a PR?

Comment (samsonasik) [reply rules/CodeQuality/Rector/FuncCall/ChangeArrayPushToArrayAssignRector.php:71]: I will try, also array_push allow to not having 2nd arg https://3v4l.org/YTXan

Comment (samsonasik) [reply rules/CodeQuality/Rector/FuncCall/ChangeArrayPushToArrayAssignRector.php:71]: https://github.com/rectorphp/rector-src/pull/2266

**Pull request merged**